draft-bormann-core-corr-clar.md

```diff
@@ -488,6 +488,69 @@ issue include:
 * <https://github.com/core-wg/corrclar/issues/9>\\
 ("Clarify/revise language around epochs in section 9.1.1 #9")
 
+## RFC 7252-9.1/11.3: Handling outdated addresses and security contexts {#amp-0rtt}
+
+INCOMPLETE:
+: Tools for mitigating these scenarios were unavailable when specified, and are now explained.
+
+PENDING.
+
+Established security contexts and established return addresses can become obsolete.
+For example, this happens when a DTLS session is resumed via CIDs, when the client's IP address changes, or when the replay window of an OSCORE context is lost and recovered through the mechanism of [Appendix B.1.2 of RFC8613].
+In those situations, a server still needs to maintain its security and and amplification mitigation properties,
+which are generally independent topics but can be addressed using the same tools.
+
+A safe option is always to reject the initial request and request confirmation.
+The RECOMMENDED way to do that is using CoAP's mechanism of sending a 4.01 (Unauthorized) response with an Echo option
+(where a subsequent request with the same Echo value proves to the server that the destination was reachable);
+clients SHOULD act to the Echo option as specified in {{?RFC9175}}.
+Tools specific to a security mechanism such as RRC ({{?I-D.ietf-tls-dtls-rrc}} may be used instead,
+but their use may depend on successful negotiation.
+
+### Amplification mitigation and return routability
+
+If it is not certain that the client is reachable on the request's sender address,
+but the response does not exceed the request's size by a factor of 3 ({{Section 2.4 of RFC9175}}, item 3),
+the server can answer the request.
+It should still include an Echo value, whose presence in the next request serves to confirm the client's address.
+
+This situation can happen at any time in OSCORE,
+or in DTLS after a CID based resumption.
+
+Verifying the client's address is not only relevant for amplification attacks
+(which addresses attacks described in {{?I-D.irtf-t2trg-amplification-attacks}})
+but also for traffic misdirection.
+{{Section 7 of ?I-D.ietf-tls-dtls-rrc}} contains a menu of options how to use RRC messages to distinguish different cases.
+An 4.01 response with Echo can perform some of the functions equivalently
+(with the Echo value taking the place of the RRC cookie),
+but does not provide a means to distinguish between non-preferred and preferred paths.
+Where that distinction matters,
+RRC provides the right tools to make it.
+
+### Replay protection
+
+If it is not certain that the request is not a replay,
+but the request handler is safe or long-term idempotent
```

Comment:
but if the same request applied later ("long-term") does the same ("idempotent"), it's not required, that the server know about the replay attack.

Reply:
It's a vague field and will need better terminology.

In 7252 (or HTTP below it), idempotent requests are requests that, once executed, have no further effect.

But the term does not distinguish between whether that's just if they're executed right after another or whether anything else may happen between their executions (maybe from the same client / user, maybe from a different one). As it is used in 7252, it's about duplication, so there it's about whether a request is processed again within EXCHANGE_LIFETIME, but that doesn't tell much about what else might have changed.

It may well be that the best course of action is to just talk about safe requests, and leave idempotency at most to a footnote.

Reply:
Since f031d5253a83d25d6d87c8ecd8192a4ac420e2bf, the statement is only about "safe" and no longer about "long-term idempotent", with just an added statement that there may be more but it's hard to tell so let's not.